client/post-editor/editor-drawer/index.jsx

```diff
@@ -112,38 +113,39 @@ const EditorDrawer = React.createClass( {
 	},
 
 	renderTaxonomies: function() {
-		var element;
-
-		if ( config.isEnabled( 'manage/custom-post-types' ) &&
-				! includes( [ 'post', 'page' ], this.props.type ) ) {
-			return (
-				<EditorDrawerTaxonomies
-					postTerms={ this.props.post && this.props.post.terms }
-				/>
+		const { type, post, site } = this.props;
+
+		// Categories & Tags
+		let categories;
+		if ( 'post' === type ) {
+			categories = (
+				<CategoriesTagsAccordion
+					site={ site }
+					post={ post } />
 			);
-		}
 
-		if ( ! this.currentPostTypeSupports( 'tags' ) ) {
-			return;
+			if ( site ) {
+				categories = (
+					<CategoryListData siteId={ site.ID }>
+						<TagListData siteId={ site.ID }>
+							{ categories }
+						</TagListData>
+					</CategoryListData>
+				);
+			}
 		}
 
-		element = (
-			<CategoriesTagsAccordion
-				site={ this.props.site }
-				post={ this.props.post } />
-		);
-
-		if ( this.props.site ) {
-			element = (
-				<CategoryListData siteId={ this.props.site.ID }>
-					<TagListData siteId={ this.props.site.ID }>
-						{ element }
-					</TagListData>
-				</CategoryListData>
+		// Custom Taxonomies
+		let taxonomies;
+		if ( config.isEnabled( 'manage/custom-post-types' ) ) {
+			taxonomies = (
+				<EditorDrawerTaxonomies
+					key="taxonomies"
```

Comment:
This shouldn't be necessary because `createFragment` assigns a key itself, right?